src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,166 @@
+import VIcon from '~/components/VIcon/VIcon'
+
+const VFilterButton = defineComponent({
+  name: 'VFilterButton',
+  components: {
+    VIcon,
+    VButton,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true:
+     * - button has no border.
+     * - on mobile, the button label is blank.
+
+     * @default false
+     */
+    isHeaderScrolled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const { i18n, store } = useContext()
+    const isMdScreen = isScreen('md')
+
+    const {
+      isFilterSidebarVisible: pressed,
+      setFilterSidebarVisibility,
+    } = useFilterSidebarVisibility({ mediaQuery: isMdScreen })
+
+    const appliedFilterTags = computed(
+      () => store.getters['search/appliedFilterTags'].length
+    )
+    const isAnyFilterApplied = computed(() => appliedFilterTags.value > 0)
+
+    const labelWithCount = (count) => {
+      return i18n.tc('header.filter-button.with-count', count)
+    }
+    // TODO(obulat): Remove buttonVariants. The aim of this object was to lower
+    // the cognitive load of remembering the variant names when debugging the
+    // button.
+    const buttonVariants = {
+      pressed: 'action-menu-pressed',
+      gray: 'action-menu-muted',
+      bordered: 'action-menu',
+      textOnly: 'action-menu-secondary',
+    }
+
+    const icon = reactive({
+      show: true,
+      needsPadding: false,
+    })
+
+    const button = reactive({
+      variant: 'action-menu',
+      label: i18n.t('header.filter-button.simple'),
+      a11yProps: 'sr-only',
+    })
+
+    const computeButtonVariant = () => {
+      let variant
+      if (pressed.value) {
+        variant = 'pressed'
+      } else {
+        if (isAnyFilterApplied.value) {
+          variant = 'gray'
+        } else {
+          variant = props.isHeaderScrolled ? 'textOnly' : 'bordered'
+        }
+      }
+      return buttonVariants[variant]
+    }
+    const computeButtonLabel = () => {
+      if (isAnyFilterApplied.value) {
+        button.label =
+          !isMdScreen.value && props.isHeaderScrolled
+            ? appliedFilterTags.value.toLocaleString('en')
+            : labelWithCount(appliedFilterTags.value)
+        button.a11yProps = ''
+        icon.show = false
+        icon.needsPadding = false
+      } else {
+        button.label = i18n.t('header.filter-button.simple')
+        button.a11yProps = isMdScreen.value ? '' : 'sr-only'
+        icon.show = true
+        icon.needsPadding = !!isMdScreen.value
+      }
+    }
+    const updateButton = () => {
+      button.variant = computeButtonVariant()
+      computeButtonLabel()
+    }
+
+    /**
+     * Since the page is rendered as mobile first, i.e. with a visually-hidden
+     * text label initially, running updateButton adds a visual label if the
+     * screen size is larger than mobile when it is mounted.
+     */
+    onMounted(() => updateButton())
+
+    watchEffect(() => {
+      updateButton()
+    })
+
```

Comment:
This can be condensed into just a `watchEffect` with the `immediate` option:

```suggestion
    /**
     * Since the page is rendered as mobile first, i.e. with a visually-hidden
     * text label initially, running updateButton adds a visual label if the
     * screen size is larger than mobile when it is mounted.
     */
    watchEffect(
      () => {
        updateButton()
      },
      { immediate: true }
    )

```

Reply:
Thank you, I didn't know about that :)